Define a keymap prefix for projectile

flexyford opened the pull request:
Starting with projectile version 1.1, the projectile (and counsel-projectile) key bindings are only available after you select a keymap prefix for them. For instance, to select C-c p as prefix (the default prior to version 1.1), you need to execute the following form:

> (define-key projectile-mode-map (kbd "C-c p") 'projectile-command-map)

Fixes #132

flexyford committed:
define a keymap prefix for projectile

Starting with projectile version 1.1, the projectile (and counsel-projectile) key bindings are only available after you select a keymap prefix for them. For instance, to select C-c p as prefix (the default prior to version 1.1), you need to execute the following form:

cowboyd reviewed (commented):
Should we update the dependency to counsel-projectile 1.1 then?

https://github.com/thefrontside/frontmacs/blob/ead96395efc4640e8922d2d48a7e6cf1f27a9b20/frontmacs-pkg.el#L13

flexyford commented:
I'm happy to give it a shot locally. I'm on `counsel-projectile-20190325.856` which I find confusing. What does that timestamp indicate? Non-stable version?

cowboyd commented:
Yeah, that represents a master snapshot from march 25th of 2019

flexyford committed:
Bump counsel-projectile to 0.3.0 stable version

Remove explicit `counsel` and `projectile` packages since these are
dependencies of `counsel-projectile`

flexyford committed:
do not use ivy--regex-fuzzy for counsel-ag

A bug in swiper is causing counsel-ag to not narrow any results using
fuzzy search. This will be fixed in an upcoming release of swiper.

https://github.com/abo-abo/swiper/issues/1907

flexyford commented:
> Should we update the dependency to counsel-projectile 1.1 then?

I see the latest as `0.3.0` on melpa stable, so I bumped to `0.3.0`

Fuzzy search was broken for `counsel-ag`  so added a fix for that as well. 

---

Everything looks good on my end after bumping counsel-projectile to latest stable `0.3.0`. I also see that `counsel-projectile` requires `counsel-0.11.0` `projectile-2.0.0`. I went ahead and removed those packages.

```
counsel-projectile is a dependency package.

Status: Installed in ‘counsel-projectile-0.3.0/’ (unsigned).
Version: 0.3.0
Commit: d64e5275c578a494102852c466a3696bde466739
Summary: Ivy integration for Projectile
Requires: counsel-0.11.0, projectile-2.0.0
Required by: frontmacs-0.2.21
Homepage: https://github.com/ericdanan/counsel-projectile
Keywords: project convenience 
Other versions: 0.3.0 (MELPA Stable), 20190325.856 (MELPA).
```

https://github.com/thefrontside/frontmacs/blob/ead96395efc4640e8922d2d48a7e6cf1f27a9b20/frontmacs-pkg.el#L9-L12

cowboyd commented on frontmacs-completion.el:
```diff
@@ -29,8 +29,10 @@
+;; Disable for counsel-ag since it's not narrowing results with fuzzy.
 (setq ivy-re-builders-alist
       '((swiper . ivy--regex-plus)
+        (counsel-ag . ivy--regex-plus)
```
This has been annoying me so much! THANK YOU 👍 👍 🙏

flexyford replied:
Ugh, yea same here. As stated in b88e612

> A bug in swiper is causing counsel-ag to not narrow any results using
fuzzy search. This will be fixed in an upcoming release of swiper.

The present version is `swiper@0.11.0` as of this writing. So whatever the next release is we can update the package and remove that line.

cowboyd reviewed (approved):
Jolly good show old bean!

cowboyd commented:
Well that's odd seeing as it passed on the PR https://travis-ci.org/thefrontside/frontmacs/jobs/531853721#L1033

<blockquote><div><strong><a href="https://travis-ci.org/thefrontside/frontmacs/jobs/531853721#L1033">Travis CI - Test and Deploy Your Code with Confidence</a></strong></div></blockquote>
